serveradmin/serverdb/query_committer.py

```diff
@@ -434,24 +434,26 @@ def _validate_readonly(attribute_lookup, changes, servers):
     return violations
 
 
-def _validate_regexp(changes, servers, servertype_attributes):
+def _validate_regexp(changes, servers, attribute_lookup):
     for attribute_changes in changes:
         object_id = attribute_changes['object_id']
-        server = servers[object_id]
         for attribute_id, change in attribute_changes.items():
-            sa = servertype_attributes[server['servertype']].get(attribute_id)
-            if not sa or not sa.regexp:
+            if attribute_id in Attribute.specials:
+                continue
```

Comment:
Shouldn't we validate special attributes as well?

Reply:
attribute_lookup does not contain special attributes. As far as I understand special attributes are regexp-validated by the django ORM via the 'validators' defined in the serverdb models.py. I'm not sure if we maybe bypass these validators here though.. 
Think the `if not sa or not sa.regexp` had the same effect before.